### ISSUE TITLE
Preference Transactions Failing when Run in Parallel

### ISSUE THREAD

**Issue by colin-grant-work** [eclipse-theia/theia#10824]
> A user posted this, and although the answer works, the idea of the PR linked there was actually to reliably _allow_ rapid preference setting, rather than causing errors when preferences are set rapidly.

I'm currently facing this issue while setting CustomPreference. I'm having a `CustomDialog` which sets multiple preference using for loop upon clicking OK. The first three preferences sets just fine. However, post that it throws the following error. This dialog used to work fine with previous theia version. Any help would be appreciated. 

Thanks 

```
Canceled: Canceled
    at new Transaction (http://localhost:3000/vendors-node_modules_theia_preferences_lib_browser_preference-frontend-module_js.bundle.js:1017:46)
    at new PreferenceTransaction (http://localhost:3000/vendors-node_modules_theia_preferences_lib_browser_preference-frontend-module_js.bundle.js:1123:29)
    at _createInstance (http://localhost:3000/bundle.js:78216:12)
    at Object.resolveInstance (http://localhost:3000/bundle.js:78236:18)
    at http://localhost:3000/bundle.js:78331:42
    at Object.resolve (http://localhost:3000/bundle.js:78355:12)
    at http://localhost:3000/bundle.js:77374:37
    at Container.__webpack_modules__.../../node_modules/inversify/lib/container/container.js.Container._get (http://localhost:3000/bundle.js:77365:44)
    at Container.__webpack_modules__.../../node_modules/inversify/lib/container/container.js.Container.get (http://localhost:3000/bundle.js:77282:21)
    at UserPreferenceProvider.transactionFactory (http://localhost:3000/vendors-node_modules_theia_preferences_lib_browser_preference-frontend-module_js.bundle.js:952:22)
```

```
                Object.keys(values).forEach(async (ele) => {
                    await preferenceService.set('instance.search.' + ele, values[ele], PreferenceScope.Workspace)
                })
```

_Originally posted by @maheshrajrp in https://github.com/eclipse-theia/theia/discussions/10744_

**Comment by colin-grant-work** [comment]
@msujew, you mentioned having had the same problem - do you recall any details?

**Comment by msujew** [comment]
@colin-grant-work [this](https://github.com/arduino/arduino-ide/blob/51da3c0668f0357ab4666e6b9d290e32b49fc277/arduino-ide-extension/src/browser/dialogs/settings/settings.tsx#L256-L314) was the old code from the Arduino-IDE, which led to the race condition (at least I assume it to be a race condition).

I'm not fully sure anymore what I saw during debugging, before I simply decided to run everything in sequence, but I think some variables inside of some preference provider were set outside of the transaction, which led to the preferences not being applied correctly.

By the way, have you noticed this message when quickly changing preferences (like quickly changing the color theme)?

![image](https://user-images.githubusercontent.com/4377073/158280037-bfb32ae5-6239-4b06-878e-28ffe271dda4.png)

I didn't catch it during review, but it appears almost every time when changing color/icon themes.
